satpy/readers/seviri_l1b_nc.py

```diff
@@ -159,49 +166,44 @@ def get_area_def(self, dataset_id):
         return area
 
     def get_area_extent(self, dsid):
-        if dsid.name != 'HRV':
-
-            # following calculations assume grid origin is south-east corner
-            # section 7.2.4 of MSG Level 1.5 Image Data Format Description
-            origins = {0: 'NW', 1: 'SW', 2: 'SE', 3: 'NE'}
-            grid_origin = self.nc.attrs['vis_ir_grid_origin']
-            grid_origin = int(grid_origin, 16)
-            if grid_origin != 2:
-                raise NotImplementedError(
-                    'Grid origin not supported number: {}, {} corner'
-                    .format(grid_origin, origins[grid_origin])
-                )
-
-            center_point = 3712/2
-
-            column_step = self.nc.attrs['vis_ir_column_dir_grid_step'] * 1000.0
-
-            line_step = self.nc.attrs['vis_ir_line_dir_grid_step'] * 1000.0
-            # section 3.1.4.2 of MSG Level 1.5 Image Data Format Description
-            earth_model = int(self.nc.attrs['type_of_earth_model'], 16)
-            if earth_model == 2:
-                ns_offset = 0  # north +ve
-                we_offset = 0  # west +ve
-            elif earth_model == 1:
-                ns_offset = -0.5  # north +ve
-                we_offset = 0.5  # west +ve
-            else:
-                raise NotImplementedError(
-                    'unrecognised earth model: {}'.format(earth_model)
-                )
-            # section 3.1.5 of MSG Level 1.5 Image Data Format Description
-            ll_c = (center_point - self.west - 0.5 + we_offset) * column_step
-            ll_l = (self.south - center_point - 0.5 + ns_offset) * line_step
-            ur_c = (center_point - self.east + 0.5 + we_offset) * column_step
-            ur_l = (self.north - center_point + 0.5 + ns_offset) * line_step
-            area_extent = (ll_c, ll_l, ur_c, ur_l)
 
+        # following calculations assume grid origin is south-east corner
+        # section 7.2.4 of MSG Level 1.5 Image Data Format Description
+        origins = {0: 'NW', 1: 'SW', 2: 'SE', 3: 'NE'}
+        grid_origin = self.nc.attrs['vis_ir_grid_origin']
+        grid_origin = int(grid_origin, 16)
+        if grid_origin != 2:
+            raise NotImplementedError(
+                'Grid origin not supported number: {}, {} corner'
+                .format(grid_origin, origins[grid_origin])
+            )
+
+        center_point = 3712/2
+
+        column_step = self.nc.attrs['vis_ir_column_dir_grid_step'] * 1000.0
+
+        line_step = self.nc.attrs['vis_ir_line_dir_grid_step'] * 1000.0
+        # section 3.1.4.2 of MSG Level 1.5 Image Data Format Description
+        earth_model = int(self.nc.attrs['type_of_earth_model'], 16)
+        if earth_model == 2:
+            ns_offset = 0  # north +ve
+            we_offset = 0  # west +ve
+        elif earth_model == 1:
```

Comment:
Maybe add a short comment that the data is shifted to N-W and to correct for it, we have to adjust the corners in the opposite direction, i.e. S-E

Reply:
the comment can be copied from the `seviri_l1b_native`-reader.

Reply:
Working on creating tests for this reader now

I will add the comment as suggested by sjoro

Reply:
comment added